File: client/src/types/JsonModels/Auth/UserAccessModel.ts
import type { UserRole } from "../Enums/UserRole";

export interface UserAccessModel
{
	id: number;
	personId?: number;
	email: string;
	firstName?: string;
	lastName?: string;
	personDisplayName?: string;
	isActive: boolean;
	roles: UserRole[];
}

File: server/Bank.Core/JsonModels/Auth/UserAccessModel.cs
namespace Bank.Core.JsonModels.Auth;

public class UserAccessModel
{
    public long Id { get; set; }
    public long? PersonId { get; set; }
    public string Email { get; set; } = string.Empty;
    public string? FirstName { get; set; }
    public string? LastName { get; set; }
    public string? PersonDisplayName { get; set; }
    public bool IsActive { get; set; }
    public IReadOnlyCollection<UserRole> Roles { get; set; } = [];
}
